zebra-consensus/src/chain.rs

```diff
@@ -58,32 +62,40 @@ where
         Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        match (self.checkpoint.poll_ready(cx), self.block.poll_ready(cx)) {
-            // First, fail if either service fails.
-            (Poll::Ready(Err(e)), _) => Poll::Ready(Err(VerifyChainError::Checkpoint(e))),
-            (_, Poll::Ready(Err(e))) => Poll::Ready(Err(VerifyChainError::Block(e))),
-            // Second, we're unready if either service is unready.
-            (Poll::Pending, _) | (_, Poll::Pending) => Poll::Pending,
-            // Finally, we're ready if both services are ready and OK.
-            (Poll::Ready(Ok(())), Poll::Ready(Ok(()))) => Poll::Ready(Ok(())),
-        }
+        self.checkpoint.poll_ready(cx);
+        self.block.poll_ready(cx);
+
+        Poll::Ready(Ok(()))
```

Comment:
```suggestion
        // Correctness:
        //
        // We can't call `poll_ready` on the block and checkpoint verifiers here, because that leaks buffer slots.
        // See #1593 for details.
        Poll::Ready(Ok(()))
```